protostar/cheatable_starknet/cheatable_entry_point.py

```diff
@@ -276,15 +277,29 @@ def execute(
         else:
             sync_state = state
 
+        raw_contract_address = Address.from_user_input(self.contract_address)
+
         self.cheaters.expects.unregister_expected_call(
-            contract_address=Address.from_user_input(self.contract_address),
+            contract_address=raw_contract_address,
             function_selector=self.entry_point_selector,
             calldata=self.calldata,
         )
 
-        return super().execute(
+        result = super().execute(
```

Comment:
If the call is to be mocked, it's unnecessary (and incorrect) for the call to be even made.